docs/rst/tutorials/micro/user_api/user_api_actions.rst

```diff
@@ -0,0 +1,388 @@
+    void * context)
+  {
+    (void) context;
+
+    example_interfaces__action__Fibonacci_SendGoal_Request * request =
+      (example_interfaces__action__Fibonacci_SendGoal_Request *) goal_handle->ros_goal_request;
+
+    example_interfaces__action__Fibonacci_GetResult_Response * result =
+      (example_interfaces__action__Fibonacci_GetResult_Response *) ros_result_response;
+
+    printf(
+      "Goal Result (order: %d) [ ",
+      request->goal.order
+    );
+
+    if (result->status == GOAL_STATE_SUCCEEDED) {
+      for (size_t i = 0; i < result->result.sequence.size; i++) {
+        printf("%d ", result->result.sequence.data[i]);
+      }
+    } else if (result->status == GOAL_STATE_CANCELED) {
+      printf("CANCELED ");
+    } else {
+      printf("ABORTED ");
+    }
+
+    printf("]\n");
+  }
+
+During the goal execution, the action client can request a goal cancellation:
+
+.. code-block:: c
+
+  // Send goal cancel request
+  rclc_action_send_cancel_request(goal_handle);
+
+And define a callback for the cancel request response:
+
+.. code-block:: c
+
+  // Function prototype:
+  void (* cancel_request_callback_t)(rclc_action_goal_handle_t *, bool, void *);
+
+  void cancel_request_callback(
+    rclc_action_goal_handle_t * goal_handle,
+    bool cancelled,
+    void * context)
+  {
+    example_interfaces__action__Fibonacci_SendGoal_Request * request =
+      (example_interfaces__action__Fibonacci_SendGoal_Request *) goal_handle->ros_goal_request;
+
+    printf(
+      "Goal cancel request (order: %d): %s\n",
+      request->goal.order,
+      cancelled ? "Accepted" : "Rejected");
+  }
+
+Finally, all the callbacks shall be exposed to the executor when the action client is added:
+
+.. code-block:: c
+
+  #define MAX_FIBONACCI_ORDER 50
+
+  // Action message objects
+  example_interfaces__action__Fibonacci_FeedbackMessage ros_feedback;
+  example_interfaces__action__Fibonacci_GetResult_Response ros_result_response;
+
+  // Init message memory with expected sizes
+  ros_feedback.feedback.sequence.capacity = MAX_FIBONACCI_ORDER;
+  ros_feedback.feedback.sequence.size = 0;
+  ros_feedback.feedback.sequence.data = (int32_t *) malloc(
+    ros_feedback.feedback.sequence.capacity * sizeof(int32_t));
+
+  ros_result_response.result.sequence.capacity = MAX_FIBONACCI_ORDER;
+  ros_result_response.result.sequence.size = 0;
+  ros_result_response.result.sequence.data = (int32_t *) malloc(
+    ros_result_response.result.sequence.capacity * sizeof(int32_t));
+
+  // Add action client to the executor ('executor' and 'action_client' are available and already initialized)
+  rclc_executor_add_action_client(
+    &executor,
+    &action_client,
+    NUMBER_OF_SIMULTANEOUS_HANDLES,
+    &ros_result_response,
+    &ros_feedback,
+    goal_request_callback,
+    feedback_callback,
+    result_request_callback,
+    cancel_request_callback,
+    NULL
+  );
+
+  // Spin executor to receive requests
+  rclc_executor_spin(&executor);
```

Comment:
```suggestion
```

Missed this one, the client doesn't receive requests

Reply:
aa728e43bfa90e587d4cee96732918973302f6c1